docs/site/migration/auth/passport.md

```diff
@@ -10,8 +10,378 @@ permalink: /doc/en/lb4/migration-auth-passport.html
 Missing instructions for your LoopBack 3 use case? Please report a [Migration docs issue](https://github.com/strongloop/loopback-next/issues/new?labels=question,Migration,Docs&template=Migration_docs.md) on GitHub to let us know.
 " %}
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3958](https://github.com/strongloop/loopback-next/issues/3958)
-" %}
+# Migrating Passport Strategies
+
+This page guides on migrating lb3 apps that used
+[Passport Strategies](http://www.passportjs.org/) for authentication usecases.
+Before following this guide, please know more about the
+[@loopback/authentication-passport](https://loopback.io/doc/en/lb4/Authentication-passport.html)
+package.
+
+## Differences between LoopBack 3 and LoopBack 4
+
+In LoopBack 3, routes can be configured explicitly as authentication providers
+using express style passport strategies middleware. Also the
+[lb3 passport component](https://github.com/strongloop/loopback-component-passport)
+helped with implicit authentication configuration using json files. It had
+built-in model classes to search users and persist user identities.
+
+In LoopBack 4, authentication endpoints are configured in controllers and the
+`@authenticate` decorator tells which passport strategy to configure for that
+API route. Also the
+[@loopback/authentication-passport](https://loopback.io/doc/en/lb4/Authentication-passport.html)
+package is necessary to bridge between passport strategies and the
+authentication design of lb4.
+
+## An example passport login app
+
+To demonstrate how to implement passport strategies in LoopBack 4 and migrate
+LB3 apps
+using[loopback-component-passport](https://github.com/strongloop/loopback-component-passport),
+a
+[passport-login](https://github.com/strongloop/loopback-next/tree/master/examples/passport-login)
+example app is now available.
+
+This example is migrated from
+[loopback-example-passport](https://github.com/strongloop/loopback-example-passport),
+It demonstrates how to use the LoopBack 4 features (like @authenticate
+decorator, strategy providers, etc) with passport strategies. It includes the
+oauth2 strategies to interact with external OAuth providers like facebook,
+google,etc as well as local and basic strategies.
+
+Take a look at the test cases of the
+[example app](https://github.com/strongloop/loopback-next/tree/master/examples/passport-login)
+and the
+[mock social app for testing](https://github.com/strongloop/loopback-next/tree/master/extensions/authentication-passport/src/__tests__/acceptance/fixtures)
+
+You can use this example to see how to,
+
+- Log in or Sign up into a LoopBack App using passport strategy modules
+- Log in via external apps like Facebook or link those external profiles with a
+  LoopBack user (for example, a LoopBack user can have associated
+  facebook/google accounts to retrieve pictures).
```

Comment:
```suggestion
  Facebook/Google accounts to retrieve pictures).
```

and for any occurrences of `Facebook` and `Google`.

Reply:
agreed